extension/src/popup/metrics/views.ts

```diff
@@ -22,10 +23,40 @@ const routeToEventName = {
 registerHandler<AppState>(navigate, (_, a) => {
   // Awkward, but gives us types on action payload
   const action = a as ReturnType<typeof navigate>;
-  const { pathname } = action.payload.location;
+  const { pathname, search } = action.payload.location;
+
   const eventName = routeToEventName[pathname];
+
   if (!eventName) {
     throw new Error(`Didn't find a metric event name for path '${pathname}'`);
   }
-  emitMetric(eventName);
+
+  // "/sign-transaction" and "/grant-access" require additionak metrics on loaded page
+  if (search) {
+    const { transaction, domain } = getTransactionInfo(search);
+    const METRIC_OPTION_DOMAIN = {
+      domain,
+    };
+
+    if (pathname === ROUTES.grantAccess) {
+      emitMetric(eventName, METRIC_OPTION_DOMAIN);
+    }
+
+    if (pathname === ROUTES.signTransaction) {
+      const { _operations } = transaction;
+      const operationTypes = transaction._operations.map(
```

Comment:
hm, maybe we can move this operation type mapping into `getTransactionInfo`. Might be nicer to do all the data manipulation there so all this file has to do is emit the proper metric